packages/editor/src/components/rich-text/index.js

```diff
@@ -738,13 +743,48 @@ export class RichText extends Component {
 		}
 	}
 
+	/**
+	 * Applies given annotations to the given record.
+	 *
+	 * @param {Object} record The record to apply annotations to.
+	 * @param {Array} annotations The annotation to apply.
+	 * @return {Object} A record with the annotations applied.
+	 */
+	applyAnnotations( record, annotations = [] ) {
+		annotations.forEach( ( annotation ) => {
+			let { start, end } = annotation;
+
+			if ( start > record.text.length ) {
+				start = record.text.length;
+			}
+
+			if ( end > record.text.length ) {
+				end = record.text.length;
+			}
+
+			const className = 'annotation-text annotation-text-' + annotation.source;
+
+			record = applyFormat(
+				record,
+				{ type: 'core/annotation', attributes: { className } },
```

Comment:
Ideally we should have a setting in the format registration that applies the formats, rather than here in the RichText component. I guess we could do this separately, but then I hope it will get done at some point.